Makefile

```diff
@@ -30,15 +30,15 @@ help:
 
 clean: ## Make a clean source tree
 	rm -rf build *.egg-info yarn-error.log
-	rm -rf $$(find . -name dist)
+	rm -rf $$(find . -name node_modules -type d -maxdepth 3)
+	rm -rf $$(find . -name dist -type d)
+	rm -rf $$(find . -name lib -type d)
```

Comment:
For the `node_modules`, `dist` and `lib` searches, I'd feel a little more comfortable if we could replace `'.'` with the set of directories in which we expect the named directories to reside. 

Hmm, should this be more like this:
```
        rm -rf node_modules lib dist
	rm -rf $$(find packages -name node_modules -type d -maxdepth 3)
	rm -rf $$(find packages -name dist -type d)
	rm -rf $$(find packages -name lib -type d)
        rm -rf $$(find . -name __pycache__ -type d)
```
otherwise its attempting to delete a bunch of folders that are already deleted.

I'm not concerned about `__pycache__` directories in the source tree (`elyra` or `docs`), but don't want to preclude us adding a directory like `lib` someday.

Reply:
What's your concern here? These directories  (`node_modules`, `dist` and `lib`) areis generated by the build and in a `clean` target, it should be deleted from any Elyra subdirectory, no? Could you explain what do you mean when you say: `otherwise its attempting to delete a bunch of folders that are already deleted.`?

Reply:
good point @kevin-bates, i agree and have made the changes accordingly

Reply:
@lresende - a couple things.  
1. the removal starts at the root of the project - so includes docs and elyra and I don't want to build something under one of these directories - only to have it get deleted because I run `make clean`.  This is more future-thinking; there are no cases of that now, but as soon as I write 1000 lines of code under elyra/metadata/lib, then run `make clean`, I'll be a little upset. :smile:
2. The other is just efficiency, if our plan is to cut down on time, why attempt to remove dozens of directories that no longer exist.  The find is going to locate the top folder, say `node_modules`, then find, all others in node_modules, but the `rm` is going to work off everything the `find` found - which now includes all node_module directories under the top-level node_module directory. 

Thanks for making these changes Va - approving now...